sw/device/lib/testing/test_rom/test_rom.c

```diff
@@ -98,9 +98,18 @@ bool rom_test_main(void) {
   CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &reset_reasons));
 
   // Store the reset reason in retention RAM and clear the register.
-  retention_sram_get()->reset_reasons = reset_reasons;
+  volatile retention_sram_t *ret_ram = retention_sram_get();
+  ret_ram->reset_reasons = reset_reasons;
   CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
 
+  // Write 0x54534554 (ASCII: TEST) to the end of the retention SRAM creator
```

Comment:
clever!

Reply:
Thanks! I tried checking some wen registers, but thought it would introduce unnecessary coupling. I hope this will be more flexible. Another option could be to read and check something in the ROM address space, but it would break if we decide to lock it.